configure.ac

```diff
@@ -383,61 +383,167 @@ m4_ifdef([PKG_PROG_PKG_CONFIG], [
 PKG_PROG_PKG_CONFIG
 ])
 
-# HTOP_CHECK_SCRIPT(LIBNAME, FUNCTION, DEFINE, CONFIG_SCRIPT, ELSE_PART)
-m4_define([HTOP_CHECK_SCRIPT],
-[
-   if test ! -z "m4_toupper($HTOP_[$1]_CONFIG_SCRIPT)"; then
-      # to be used to set the path to ncurses*-config when cross-compiling
-      htop_config_script_libs=$(m4_toupper($HTOP_[$1]_CONFIG_SCRIPT) --libs 2> /dev/null)
-      htop_config_script_cflags=$(m4_toupper($HTOP_[$1]_CONFIG_SCRIPT) --cflags 2> /dev/null)
-   else
-      htop_config_script_libs=$([$4] --libs 2> /dev/null)
-      htop_config_script_cflags=$([$4] --cflags 2> /dev/null)
+AC_ARG_ENABLE([unicode],
+              [AS_HELP_STRING([--enable-unicode],
+                              [enable Unicode support @<:@default=yes@:>@])],
+              [],
+              [enable_unicode=yes])
+
+AC_ARG_VAR([CURSES_CFLAGS], [C compiler flags for curses; this overrides auto detected values])
+AC_ARG_VAR([CURSES_LIBS], [linker flags for curses; this overrides auto detected values])
+
+curses_pkg_names="ncurses6 ncurses5 ncurses curses"
+
+if test "x$enable_unicode" = xyes; then
+   curses_pkg_names="ncursesw6 ncursesw5 ncursesw $curses_pkg_names"
+fi
+
+AC_ARG_WITH([curses],
+            [AS_HELP_STRING([--with-curses=NAME],
+			    [select curses package NAME; e.g. ncursesw6])], [
+   case $withval in
+   yes|no)
+      AC_MSG_ERROR([invalid value for --with-curses])
+      ;;
+   *)
+      curses_pkg_names="$withval"
+      ;;
+   esac
+
+   if test "x${CURSES_CFLAGS+y}${CURSES_LIBS+y}" = xyy; then
+      AC_MSG_WARN([ignoring --with-curses value due to override])
    fi
-   htop_script_success=no
-   htop_save_CFLAGS="$AM_CFLAGS"
-   if test ! "x$htop_config_script_libs" = x; then
-      AM_CFLAGS="$AM_CFLAGS $htop_config_script_cflags"
-      AC_CHECK_LIB([$1], [$2], [
-         AC_DEFINE([$3], 1, [The library is present.])
-         LIBS="$htop_config_script_libs $LIBS "
-         htop_script_success=yes
+], [with_curses=check])
+
+# $1: "CFLAGS" for curses
+# $2: linker flags for curses
+htop_check_libcurses_capability () {
```

Comment:
This is perhaps the first time I introduce a shell function in the configure script, and I'm not sure the function naming and comment style id good enough.

1. [The Autoconf documentation](https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.72/html_node/Shell-Functions.html) states that shell function might share the same namespace as shell variables, and to prevent name clashing and maximize portability, they use `as_fn_` prefix for shell functions. I'm not sure if we need this, by using `htop_fn_` prefix when naming this function, or I can stick to the current name and hope it won't collide with other symbols in the future.
2. The style of documenting function arguments. Should I change any of this?
```sh
# $1: "CFLAGS" for curses
# $2: linker flags for curses
```

Reply:
I think using `htop_` as prefix is fine.

For the documentation, I'd probably use either both full description (compiler flags / linker flags) or variables (CFLAGS / LDFLAGS).